recipes/bzip2/all/CMakeLists.txt

```diff
@@ -1,44 +1,34 @@
 cmake_minimum_required(VERSION 3.4)
-project(bzip2 C)
+project(bzip2 LANGUAGES C)
 
 include(GNUInstallDirs)
 
-if(MSVC OR MSVC90 OR MSVC10)
-    set(MSVC ON)
-    set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
-endif()
-
-set(SOURCE_SUBFOLDER ${CMAKE_CURRENT_SOURCE_DIR}/src)
-set(BZ2_LIBRARY bz2)
-
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 option(BZ2_BUILD_EXE ON)
 
-set(BZ2_TARGETS ${BZ2_LIBRARY})
-
-add_library(${BZ2_LIBRARY}  ${SOURCE_SUBFOLDER}/blocksort.c
-                            ${SOURCE_SUBFOLDER}/bzlib.c
-                            ${SOURCE_SUBFOLDER}/compress.c
-                            ${SOURCE_SUBFOLDER}/crctable.c
-                            ${SOURCE_SUBFOLDER}/decompress.c
-                            ${SOURCE_SUBFOLDER}/huffman.c
-                            ${SOURCE_SUBFOLDER}/randtable.c
-                            ${SOURCE_SUBFOLDER}/bzlib.h
-                            ${SOURCE_SUBFOLDER}/bzlib_private.h)
-target_include_directories(${BZ2_LIBRARY} PRIVATE ${SOURCE_SUBFOLDER})
+add_library(
+    bz2
+    ${BZ2_SRC_DIR}/blocksort.c
+    ${BZ2_SRC_DIR}/bzlib.c
+    ${BZ2_SRC_DIR}/compress.c
+    ${BZ2_SRC_DIR}/crctable.c
+    ${BZ2_SRC_DIR}/decompress.c
+    ${BZ2_SRC_DIR}/huffman.c
+    ${BZ2_SRC_DIR}/randtable.c
+)
+
+target_include_directories(bz2 PUBLIC ${BZ2_SRC_DIR})
+set_target_properties(
+    bz2
+    PROPERTIES
+    PUBLIC_HEADER "${BZ2_SRC_DIR}/bzlib.h"
+    SOVERSION ${BZ2_VERSION_MAJOR}
+    VERSION ${BZ2_VERSION_STRING}
+)
+install(TARGETS bz2)
 
 if(BZ2_BUILD_EXE)
-    add_executable(${CMAKE_PROJECT_NAME} ${SOURCE_SUBFOLDER}/bzip2.c)
-    target_link_libraries(${CMAKE_PROJECT_NAME} ${BZ2_LIBRARY})
-    target_include_directories(${CMAKE_PROJECT_NAME} PRIVATE ${SOURCE_SUBFOLDER})
-    list(APPEND BZ2_TARGETS ${CMAKE_PROJECT_NAME})
+    add_executable(bzip2 ${BZ2_SRC_DIR}/bzip2.c)
+    target_link_libraries(bzip2 PRIVATE bz2)
+    install(TARGETS bzip2)
```

Comment:
don't use this form of install, it's broken for iOS, tvOS and watchOS

Reply:
Got it. Done.